examples/image_embedder/android/app/src/main/java/com/google/mediapipe/examples/imageembedder/ImageEmbedderHelper.kt

```diff
@@ -0,0 +1,136 @@
+import com.google.mediapipe.tasks.core.Delegate
+import com.google.mediapipe.tasks.vision.imageembedder.ImageEmbedder
+import com.google.mediapipe.tasks.vision.imageembedder.ImageEmbedder.ImageEmbedderOptions
+
+class ImageEmbedderHelper(
+    private val context: Context,
+    var currentDelegate: Int = DELEGATE_CPU,
+    var currentModel: Int = MODEL_SMALL,
+    var listener: EmbedderListener?
+) {
+    private var imageEmbedder: ImageEmbedder? = null
+
+    init {
+        setupImageEmbedder()
+    }
+
+    fun setupImageEmbedder() {
+        val baseOptionsBuilder = BaseOptions.builder()
+        when (currentDelegate) {
+            DELEGATE_CPU -> {
+                baseOptionsBuilder.setDelegate(Delegate.CPU)
+            }
+            DELEGATE_GPU -> {
+                baseOptionsBuilder.setDelegate(Delegate.GPU)
+            }
+        }
+        when (currentModel) {
+            MODEL_SMALL -> {
+                baseOptionsBuilder.setModelAssetPath("mobilenet_v3_small.tflite")
+            }
+            MODEL_LARGE -> {
+                baseOptionsBuilder.setModelAssetPath("mobilenet_v3_large.tflite")
+            }
+        }
+        try {
+            val baseOptions = baseOptionsBuilder.build()
+            val optionsBuilder =
+                ImageEmbedderOptions.builder().setBaseOptions(baseOptions)
+            val options = optionsBuilder.build()
+            imageEmbedder = ImageEmbedder.createFromOptions(context, options)
+        } catch (e: IllegalStateException) {
+            listener?.onError(
+                "Image embedder failed to initialize. See error logs for " +
+                        "details"
+            )
+            Log.e(
+                TAG,
+                "Image embedder failed to load model with error: " + e.message
+            )
+        } catch (e: RuntimeException) {
+            // This occurs if the model being used does not support GPU
+            listener?.onError(
+                "Image embedder failed to initialize. See error logs for " +
+                        "details", GPU_ERROR
+            )
+            Log.e(
+                TAG,
+                "Image embedder failed to load model with error: " + e.message
+            )
+        }
+    }
+
+    fun embed(firstBitmap: Bitmap, secondBitmap: Bitmap): ResultBundle? {
+        // Inference time is the difference between the system time at the start and finish of the
+        // process
+        val startTime = SystemClock.uptimeMillis()
+
+        val firstMpImage = BitmapImageBuilder(firstBitmap).build()
+        val secondMpImage = BitmapImageBuilder(secondBitmap).build()
+        imageEmbedder?.let {
+            val firstEmbed =
+                it.embed(firstMpImage).embeddingResult().embeddings().first()
+            val secondEmbed =
+                it.embed(secondMpImage).embeddingResult().embeddings().first()
+            val inferenceTimeMs = SystemClock.uptimeMillis() - startTime
+            return ResultBundle(
```

Comment:
Let's add a comment about using cosine similarities to compare two images to determine how similar they are. Basically I just want to make sure developers that aren't familiar with what cosine similarity is are able to get a quick glance that it's only used for comparisons.